proof/invariant-abstract/SchedContext_AI.thy

```diff
@@ -1365,7 +1365,7 @@ lemma tcb_sched_action_cur_sc_tcb [wp]:
   "\<lbrace>cur_sc_tcb\<rbrace> tcb_sched_action action thread \<lbrace>\<lambda>_. cur_sc_tcb\<rbrace>"
   by (wpsimp simp: tcb_sched_action_def set_tcb_queue_def)
 
-lemma reschedule_required_cur_sc_tcb [wp]:
+lemma reschedule_required_cur_sc_tcb_inv:
```

Comment:
Seems like a reasonably safe `[wp]` rule. Any specific reason for removing it? (not that it's a problem, just curious)

Reply:
Oh, I see the rule below it, yes, that is definitely better. But then we should actually delete the rule `reschedule_required_cur_sc_tcb_inv` entirely, because `reschedule_required_cur_sc_tcb` is always better to use.

Reply:
I couldn't get the nicer rule working in the proof where I did use the `inv` version, but I'll give it another shot

Reply:
Yeah it's really not nice. It will end up making me deal with `sym_refs` and `valid_sched_context`, and while I might be able to fix it, it seems like it'll be a fairly complicated proof. Is it worth pursuing this further, or leaving it as it was?

Reply:
That seems really curious, because it should make things strictly easier. Maybe it makes another lemma not apply that expects to prove a `cur_sc_tcb` that is not there any more or something. Ok to leave it in for now, we can always remove it later.

If we do leave it in, we should prove the `_inv` lemma as a weakening of `reschedule_required_cur_sc_tcb`, though (`wpsimp` should do it).